### PR TITLE
Changelog 147

### PR DESCRIPTION
**Pull request by dima74**

### CODE REVIEW

File: _posts/2021-05-17-changelog-147.markdown [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+title: "IntelliJ Rust Changelog #147"
+date: 2021-05-17 12:00:00 +0300
+---
+
+
+## New Features
+
+* [#7140] Enable [new name resolution engine](https://blog.jetbrains.com/clion/2020/12/intellij-rust-updates-for-2020-3/#new-resolve) by default
+
+* [#6456] Add WSL support. To use it, enable `org.rust.wsl` [experimental feature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features)
```
Comment (ortem) [inline _posts/2021-05-17-changelog-147.markdown:12]:
```suggestion
* [#6456] Add [Windows Subsystem for Linux (WSL)](https://www.jetbrains.com/help/idea/how-to-use-wsl-development-environment-in-product.html) support. To use it, enable `org.rust.wsl` [experimental feature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features)
```

Comment (Undin) [inline _posts/2021-05-17-changelog-147.markdown:12]: Let's also add a note that if there are both local Windows and WSL toolchains, the local one will be selected automatically. But it's possible to specify path via `\\wsl$` paths manually for now. And it will be improved in the future

Comment (Undin) [inline _posts/2021-05-17-changelog-147.markdown:12]:
```suggestion
* [#6456] Add initial WSL support. To use it, enable `org.rust.wsl` [experimental feature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features)
```

File: _posts/2021-05-17-changelog-147.markdown [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+* [#6832] Suggest target triples in command for Cargo run configuration and run anything dialog. Suggest installing the target if it is not yet installed
+
+    <img src="/assets/posts/changelog-147/suggest_target_triple.png" alt="Target triple completion in cargo run configuration" width="700px"/>
+
+* [#7098] Support basic resolve of macros 2.0 (`pub macro name() { ... }`)
+
+## Performance Improvements
+
+* [#7181] Slightly optimize cargo refresh when using new name resolution engine
```
Comment (Undin) [inline _posts/2021-05-17-changelog-147.markdown:26]: "cargo refresh" is a more internal term. Let's rephrase it somehow. You can use "project model update" term or something similat

File: _posts/2021-05-17-changelog-147.markdown [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+## Fixes
+
+* [#7174] Cancel external linter (cargo check) immediately when some file is changed
```
Comment (Undin) [inline _posts/2021-05-17-changelog-147.markdown:30]:
```suggestion
* [#7174] Cancel [external linter](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-code-analysis.html#call-external-linter) process immediately when some file is changed. Previously, in some cases, the process might hangs
```

File: _posts/2021-05-17-changelog-147.markdown [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+* New name resolution fixes:
+
+    * [#7184] Resolve procedural macros through `#[macro_use]` on extern crates
+
+    * [#7167] Fix exception when removing crate from workspace
+
+    * [#7152] Fix changing crate name
```
Comment (Undin) [inline _posts/2021-05-17-changelog-147.markdown:38]: Let's write more user-friendly description here

**Review by Undin — CHANGES_REQUESTED** [review]
@ortem suggested increasing plugin version to 0.4 because of new name resolution engine release. I think it's ok.
Let's mention it like we did it in https://intellij-rust.github.io/2020/07/13/changelog-126.html

File: _posts/2021-05-17-changelog-147.markdown [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+* [#7140] Enable [new name resolution engine](https://blog.jetbrains.com/clion/2020/12/intellij-rust-updates-for-2020-3/#new-resolve) by default
+
+* [#6456] Add WSL support. To use it, enable `org.rust.wsl` [experimental feature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features)
+
+* [#7102], [#7165] Annotate unknown cfg attributes and provide a quick-fix. This corresponds to the [E0537](https://doc.rust-lang.org/error-index.html#E0537) compiler error. (by [@Kobzol])
+
+  {% include gif-img.html path="/assets/posts/changelog-147/unknown_cfg_predicate" w="700px" %}
+
+* [#6832] Suggest target triples in command for Cargo run configuration and run anything dialog. Suggest installing the target if it is not yet installed
```
Comment (ortem) [inline _posts/2021-05-17-changelog-147.markdown:18]:
```suggestion
* [#6832] Complete target triples in [Cargo run configuration](https://plugins.jetbrains.com/plugin/8182-rust/docs/cargo-command-configuration.html#cargo-command-config) and [`Run Anything`](https://plugins.jetbrains.com/plugin/8182-rust/docs/cargo-command-configuration.html#run-command). Suggest installing the target if it is not yet installed
```

File: _posts/2021-05-17-changelog-147.markdown (hunk 1 above)
Comment (Undin) [inline _posts/2021-05-17-changelog-147.markdown:12]: Also, don't forget to mention that support is available only since 2021.1 IDE versions

File: _posts/2021-05-17-changelog-147.markdown [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+    <img src="/assets/posts/changelog-147/suggest_target_triple.png" alt="Target triple completion in cargo run configuration" width="700px"/>
+
+* [#7098] Support basic resolve of macros 2.0 (`pub macro name() { ... }`)
```
Comment (ortem) [inline _posts/2021-05-17-changelog-147.markdown:22]:
```suggestion
* [#7098] Support basic name resolution of [macros 2.0](https://rust-lang.github.io/rfcs/1584-macros.html) (`pub macro name() { ... }`)
```

File: _posts/2021-05-17-changelog-147.markdown (hunk 4 above)
Comment (ortem) [inline _posts/2021-05-17-changelog-147.markdown:38]:
```suggestion
    * [#7152] Fix unexpected name resolution after changing a crate name
```

File: _posts/2021-05-17-changelog-147.markdown [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+## Fixes
+
+* [#7174] Cancel external linter (cargo check) immediately when some file is changed
+
+* New name resolution fixes:
+
+    * [#7184] Resolve procedural macros through `#[macro_use]` on extern crates
+
+    * [#7167] Fix exception when removing crate from workspace
```
Comment (ortem) [inline _posts/2021-05-17-changelog-147.markdown:36]:
```suggestion
    * [#7167] Fix exception when removing a crate from the workspace
```

File: _posts/2021-05-17-changelog-147.markdown [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+    * [#7152] Fix changing crate name
+
+* [#7178] Take into account enum's lifetimes in `Needless Lifetimes` inspection
+
+* [#7063] Now `Can't find crate` annotation can be suppressed, highlights only unknown identifier and highlighting style is `unknown symbol` instead of general `error`
```
Comment (ortem) [inline _posts/2021-05-17-changelog-147.markdown:42]:
```suggestion
* [#7063] Now `Can't find crate` annotation highlights only unknown identifiers and can be [suppressed](https://www.jetbrains.com/help/idea/disabling-and-enabling-inspections.html#suppress-in-editor). Also, its highlighting style was changed to `unknown symbol` (instead of `error`)
```
[minor] maybe rephrase it to something like this, in order to improve readability

**Review by ortem — APPROVED** [review]

File: _posts/2021-05-17-changelog-147.markdown [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+    * Properly support [textual scoping](https://doc.rust-lang.org/reference/macros-by-example.html#textual-scope) when resolving macros
+    * Resolve of multi-segment macro paths and macro calls inside files included by `include!`
+    * Resolve of [macros 2.0](https://github.com/rust-lang/rfcs/blob/master/text/1584-macros.md)
+    * Support file-level `#![macro_use]` and `cfg` attributes
+    * Find all possible `impl` blocks for a type alias. Previously, for performance reasons, they were processed only when the number of aliases for the corresponding type didn’t exceed 10. Now the crates that heavily use type aliases, such as [nalgebra](https://crates.io/crates/nalgebra) or [winapi](https://crates.io/crates/winapi), can be resolved properly
+
+  Note that the new algorithm works for top-level items only, so local variables and type inference functionality won’t be affected. If you run into an issue with the new engine, you can switch it back to the old one in the plugin settings `Settings | Languages & Frameworks | Rust | Use new name resolution engine`
```
Comment (Undin) [inline _posts/2021-05-17-changelog-147.markdown:20]:
```suggestion
  Note that the new algorithm works for top-level items only, so local variables and type inference functionality won’t be affected. If you run into an issue with the new engine, you can switch it back to the old one via `Use new name resolution engine` option in `Settings | Languages & Frameworks | Rust` settings
```

File: _posts/2021-05-17-changelog-147.markdown [hunk 10]
```diff
@@ -0,0 +1,80 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #147"
+date: 2021-05-17 12:00:00 +0300
```
Comment (Undin) [inline _posts/2021-05-17-changelog-147.markdown:4]:
```suggestion
date: 2021-05-18 12:00:00 +0300
```

File: _posts/2021-05-17-changelog-147.markdown [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+  Note that the new algorithm works for top-level items only, so local variables and type inference functionality won’t be affected. If you run into an issue with the new engine, you can switch it back to the old one in the plugin settings `Settings | Languages & Frameworks | Rust | Use new name resolution engine`
+
+* [#6456] Add initial [Windows Subsystem for Linux (WSL)](https://www.jetbrains.com/help/idea/how-to-use-wsl-development-environment-in-product.html) support. It is available since 2021.1, and can be enabled using `org.rust.wsl` [experimental feature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features). Note that if there are both local Windows and WSL toolchains, the local one will be selected automatically. You can manually specify `\\wsl$` paths in `Settings | Languages & Frameworks | Rust` to use WSL toolchain, and we plan to improve it in the future
```
Comment (Undin) [inline _posts/2021-05-17-changelog-147.markdown:22]:
```suggestion
* [#6456] Add initial [Windows Subsystem for Linux (WSL)](https://www.jetbrains.com/help/idea/how-to-use-wsl-development-environment-in-product.html) support. It is available since 2021.1, and can be enabled using `org.rust.wsl` [experimental feature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features). Note that if there are both local Windows and WSL toolchains, the local one will be selected automatically. You can manually specify desired WSL toolchain location with `\\wsl$` path in `Settings | Languages & Frameworks | Rust` settings. We plan to improve toolchain selection in the future
```

**Review by Undin — APPROVED** [review]
LGTM except minor comments
